GCloud DNS updates failing

Since merging #47 GCloud updates fail with:

```
Exception in thread "Thread-77" com.google.cloud.dns.DnsException: Invalid value for 'entity.change.additions[0].rrdata': '<empty>'
	at com.google.cloud.dns.spi.DefaultDnsRpc.translate(DefaultDnsRpc.java:183)
	at com.google.cloud.dns.spi.DefaultDnsRpc.applyChangeRequest(DefaultDnsRpc.java:316)
	at com.google.cloud.dns.DnsImpl$9.call(DnsImpl.java:301)
	at com.google.cloud.dns.DnsImpl$9.call(DnsImpl.java:298)
	at com.google.cloud.RetryHelper.doRetry(RetryHelper.java:179)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:244)
	at com.google.cloud.dns.DnsImpl.applyChangeRequest(DnsImpl.java:297)
	at com.google.cloud.dns.Zone.applyChangeRequest(Zone.java:178)
	at in.tazj.k8s.letsencrypt.acme.CloudDnsResponder.updateCloudDnsRecord(CloudDnsResponder.java:109)
	at in.tazj.k8s.letsencrypt.acme.CloudDnsResponder.removeChallengeRecord(CloudDnsResponder.java:49)
	at in.tazj.k8s.letsencrypt.acme.CertificateRequestHandler.lambda$prepareDnsChallenge$1(CertificateRequestHandler.java:187)
	at in.tazj.k8s.letsencrypt.acme.CertificateRequestHandler.authorizeDomain(CertificateRequestHandler.java:85)
	at in.tazj.k8s.letsencrypt.acme.CertificateRequestHandler.lambda$requestCertificate$0(CertificateRequestHandler.java:59)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1374)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734)
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)
	at in.tazj.k8s.letsencrypt.acme.CertificateRequestHandler.requestCertificate(CertificateRequestHandler.java:59)
	at in.tazj.k8s.letsencrypt.kubernetes.ServiceManager.handleCertificateRequest(ServiceManager.java:108)
	at java.util.Optional.ifPresent(Optional.java:159)
	at in.tazj.k8s.letsencrypt.kubernetes.ServiceManager.lambda$reconcileService$0(ServiceManager.java:58)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
{
  "code" : 400,
  "errors" : [ {
    "domain" : "global",
    "message" : "Invalid value for 'entity.change.additions[0].rrdata': '<empty>'",
    "reason" : "invalid"
  } ],
  "message" : "Invalid value for 'entity.change.additions[0].rrdata': '<empty>'"
}
	at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:145)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:113)
	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:40)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:321)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1056)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:419)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
	at com.google.cloud.dns.spi.DefaultDnsRpc.applyChangeRequest(DefaultDnsRpc.java:314)
	... 29 more
```

This is only for the snapshot builds, not stable.

--- a/src/main/java/in/tazj/k8s/letsencrypt/acme/CloudDnsResponder.java
+++ b/src/main/java/in/tazj/k8s/letsencrypt/acme/CloudDnsResponder.java
@@ -77,34 +77,30 @@ public class CloudDnsResponder implements DnsResponder {
 
     val zone = matchingZone.get();
     val fqdnRecord = determineFqdnRecord(recordName);
-    val recordSetBuilder = RecordSet
-        .newBuilder(fqdnRecord, Type.TXT)
-        .setTtl(1, MINUTES);
 
-    final RecordSet recordSet;
-    if (changeType.equals(ADD)) {
-      recordSet = recordSetBuilder
-          .addRecord(challengeDigest)
-          .build();
-    } else {
-      recordSet = recordSetBuilder
-          .removeRecord(challengeDigest)
-          .build();
-    }
+    val recordSet = RecordSet
+        .newBuilder(fqdnRecord, Type.TXT)
+        .setTtl(1, MINUTES)
+        .addRecord(challengeDigest)
+        .build();
 
     val changeBuilder = ChangeRequestInfo.newBuilder();
 
-    // Verify there is no existing record / overwrite it if there is.
-    Iterator<RecordSet> recordSetIterator = zone.listRecordSets().iterateAll();
-    while (recordSetIterator.hasNext()) {
-      RecordSet current = recordSetIterator.next();
-      if (recordSet.getName().equals(current.getName()) &&
-          recordSet.getType().equals(current.getType())) {
-        changeBuilder.delete(current);
+    if (changeType.equals(ADD)) {
+      // Verify there is no existing record / overwrite it if there is.
+      Iterator<RecordSet> recordSetIterator = zone.listRecordSets().iterateAll();
+      while (recordSetIterator.hasNext()) {
+        val current = recordSetIterator.next();
+        if (recordSet.getName().equals(current.getName()) &&
+            recordSet.getType().equals(current.getType())) {
+          changeBuilder.delete(current);
+        }
       }
-    }
 
-    changeBuilder.add(recordSet);
+      changeBuilder.add(recordSet);
+    } else {
+      changeBuilder.delete(recordSet);
+    }
 
     return zone.applyChangeRequest(changeBuilder.build());
   }

